clients/grailfiltersegments/filtersegment_test.go

```diff
@@ -0,0 +1,476 @@
+// @license
+// Copyright 2024 Dynatrace LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grailfiltersegments_test
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/clients/grailfiltersegments"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/testutils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func Test_List(t *testing.T) {
+	t.Run("base case", func(t *testing.T) {
+		apiResponse := `{
+  "filterSegments": [
+    {
+      "uid": "QElQbQcjq3S",
+      "name": "filter_name",
+      "isPublic": false,
+      "owner": "userUUID",
+      "version": 1
+    }
+  ]
+}`
+		expected := `[
+    {
+      "uid": "QElQbQcjq3S",
+      "name": "filter_name",
+      "isPublic": false,
+      "owner": "userUUID",
+      "version": 1
+    }
+  ]`
+
+		mockClient := grailfiltersegments.NewMockclient(gomock.NewController(t))
+		mockClient.EXPECT().
+			List(gomock.Any()).
+			Return(&http.Response{
+				StatusCode: http.StatusOK,
+				Body:       io.NopCloser(strings.NewReader(apiResponse)),
+			}, nil)
+
+		fsClient := grailfiltersegments.NewTestClient(mockClient)
+		actual, err := fsClient.List(context.Background())
+
+		require.NoError(t, err)
+		assert.JSONEq(t, expected, string(actual.Data))
+	})
+}
+
+func Test_Get(t *testing.T) {
+	t.Run("Get - no ID given", func(t *testing.T) {
```

Comment:
The naming should just be `No ID`. The rest of the context follows from the test-name. Same for the other cases

```suggestion
	t.Run("No ID", func(t *testing.T) {
```